fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -869,6 +869,18 @@ class BrowserRobot {
             getStringResource(R.string.open_in_app_cfr_negative_button_text),
         ).click()
 
+    fun clickTakeSurveyButton() {
+        val button = mDevice.findObject(
+            UiSelector().text(
+                getStringResource(
+                    R.string.preferences_take_survey,
+                ),
+            ),
+        )
+        button.waitForExists(waitingTimeShort)
+        button.click()
```

Comment:
Let's split this function up so we verify the existence first. In class BrowserRobot, I'd add: 

```Kotlin
fun verifySurveyButton() { 
     val button = mDevice.findObject(
            UiSelector().text(
                getStringResource(
                    R.string.preferences_take_survey,
                ),
            ),
        )
    assertTrue(button.waitForExists(waitingTimeshort))
} 
```
and add an accessor at the bottom of the file
```Kotlin
private val surveyButton = 
    itemContainingText(getStringResource(R.string.preferences_take_survey))
```
and in Class Transition we can write a new function to handle the click transition

```Kotlin
fun clickSurveyButton(interact: BrowserRobot.() -> unit): BrowserRobot.Transition {
     surveyButton().waitforExists(waitingTimeshort)
     surveyButton().click()

      BrowserRobot().interact()
      return BrowserRobot.Transition()
}
```

and then back in the test you should be able to call 

```Kotlin
 browserScreen {
            verifySurveyButton()
            clickSurveyButton()
        }
```
So in that regard we can verify things upfront

(@sv-ohorvath does this make sense?)

Reply:
We usually use the Transition class if we need to switch to another class (say the survey button would open another screen than the browser). In this case, if we stay on the same browser page, we don't need the transition, so the clickSurveyButton method can live inside BrowserRobot class.